com.playeveryware.eos/Runtime/Core/EOSManager.cs

```diff
@@ -491,40 +491,97 @@ private void InitializeOverlay(IEOSCoroutineOwner coroutineOwner)
+                    bool sandboxDefined = false;
+                    SandboxId sandboxFromCommandLine = SandboxId.FromString(epicArgs.epicSandboxID);
+                    foreach (var namedSandbox in productConfig.Environments.Sandboxes)
                     {
-                        continue;
+                        if (namedSandbox.Value.Equals(sandboxFromCommandLine))
+                        {
+                            Debug.Log($"{namedSandbox} selected as sandbox.");
+                            sandboxDefined = true;
+                            break;
+                        }
                     }
 
-                    PlatformManager.GetPlatformConfig().deployment.SandboxId = sandbox.Value;
-                    sandboxOverridden = true;
-                    Debug.Log($"SandboxID was overridden to be \"{sandbox.Value}\" by a command line parameter.");
-                    break;
+                    PlatformManager.GetPlatformConfig().deployment.SandboxId = sandboxFromCommandLine;
+
+                    if (!sandboxDefined)
+                    {
+                        Debug.LogWarning(
+                            $"Sandbox Id \"{sandboxFromCommandLine}\" was " +
+                            $"provided on the command line, but was not " +
+                            $"found in the product config. Attempting to use " +
+                            $"it regardless.");
+                    }
                 }
 
-                if (!sandboxOverridden)
+                if (!string.IsNullOrEmpty(epicArgs.epicDeploymentID))
                 {
-                    Debug.LogWarning(
-                        $"The SandboxID " +
-                        $"\"{epicArgs.epicSandboxID}\" specified by " +
-                        $"command line argument is not a valid id. " +
-                        $"Defaulting to SandboxID " +
-                        $"\"{PlatformManager.GetPlatformConfig().deployment.SandboxId}\" " +
-                        $"defined in the configuration.");
+                    bool deploymentDefined = false;
+
+                    foreach (var namedDeployment in productConfig.Environments.Deployments)
+                    {
+                        // Check for equality regardless of case - and
+                        // regardless of whether the dashes are included in the
+                        // Guid for the purposes of comparison.
+                        if (namedDeployment.Value.DeploymentId.ToString().Equals(epicArgs.epicDeploymentID,
+                                StringComparison.OrdinalIgnoreCase) ||
+                            namedDeployment.Value.DeploymentId.ToString("N").Equals(epicArgs.epicDeploymentID,
+                                StringComparison.OrdinalIgnoreCase))
+                        {
+                            Debug.Log($"{namedDeployment} selected as deployment.");
+                            deploymentDefined = true;
```

Comment:
nit; If we've found a match, then we should `break` the foreach loop.

Reply:
Excellent observation - I've corrected the implementation. Marking the conversation as resolved.